### PR TITLE
FEATURE: Live updates when comments are created.

### PR DESCRIPTION
**Pull request by tgxworld**

### CODE REVIEW

File: test/javascripts/acceptance/question-answer-test.js [hunk 1]
```diff
@@ -568,7 +574,7 @@ acceptance("Discourse Question Answer - logged in user", function (needs) {
   // Skip message bus tests but keep it around for development use. We currently do not have a reliable way to wait for
   // widgets to re-render and this results in all kinds of timing issues. For example, this test passes in the browser
   // but fails when ran in headless mode.
-  skip("receiving user post voted message where current user removed their vote", async function (assert) {
```
Comment (tgxworld) [inline test/javascripts/acceptance/question-answer-test.js:571]: I learnt about `await settled()` today which resolved the problem we were seeing previously.

File: app/serializers/question_answer_comment_serializer.rb [hunk 2]
```diff
@@ -25,7 +25,7 @@ def user_voted
     if @comments_user_voted
       @comments_user_voted[object.id]
     else
-      object.votes.exists?(user: scope.user)
+      !!(scope && object.votes.exists?(user: scope.user))
```
Comment (martin-brennan) [inline app/serializers/question_answer_comment_serializer.rb:28]: This `!!` syntax always feels bad to me...can't we just do `scope.present? && object.votes.exists?(user: scope.user)`

File: lib/question_answer/comment_creator.rb [hunk 3]
```diff
@@ -8,12 +8,26 @@ def self.create(attributes)
...
+            publish_changes(qa_comment)
+          end
         end
       end
 
       qa_comment
     end
 
+    def self.publish_changes(qa_comment)
+      Scheduler::Defer.later "Publish new Q&A comment" do
```
Comment (martin-brennan) [inline lib/question_answer/comment_creator.rb:22]: Just wondering why this needs to be deferred?

**Review by martin-brennan — APPROVED** [review]
Looks good, just a couple of questions but I won't hold up the PR for them

File: lib/question_answer/comment_creator.rb (hunk 3 above)
Comment (tgxworld) [reply lib/question_answer/comment_creator.rb:22]: I did this for performance reasons. Publishing a message is not urgent so I don't want to add the overhead to the request.

File: app/serializers/question_answer_comment_serializer.rb (hunk 2 above)
Comment (tgxworld) [reply app/serializers/question_answer_comment_serializer.rb]: Good point :+1:

File: lib/question_answer/comment_creator.rb (hunk 3 above)
Comment (martin-brennan) [reply lib/question_answer/comment_creator.rb:22]: Ah okay, no problem, I guess a lot of the time these type of publishes are already within jobs etc. so I just haven't noticed before.

**Comment by tgxworld** [comment]
Thank you @martin-brennan

**Pull request merged**